x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/support/SecurityIndexManagerIntegTests.java

```diff
@@ -70,6 +71,8 @@ protected void doRun() throws Exception {
 
         assertThat(exceptions, Matchers.empty());
         assertEquals(futures.size(), numRequests * numThreads);
+        // TODO: We could also just assert future.actionGet does not throw since either created
+        //       or updated is sufficient for the test purpose, i.e. security index is created.
         for (ActionFuture<PutUserResponse> future : futures) {
             assertTrue(future.actionGet().created());
         }
```

Comment:
I have to add this comment to be able to comment this piece of code on the GitHub page. Since the test just needs to ensure security index is created when there are concurrent operations, either a "created" or "updated" user is sufficient. If we make this change, it has the benefit to not fail the test if the same edge case (#73779) happens. Note this change does not replace the reduction of concurrency because large number of threads can fail at prepration phase which is before the requests can be sent off at all.

Reply:
I think we can make that change, given what the test is really trying to do.
It's unfortunate that PutUser might sometimes given a misleading response, but it's a natural consequence consequence consequence of not being able to guarantee exactly-once-delivery. Sometime an internal retry will produce a no-op update instead of a create.
But the point is that the index & alias are created and the Put operation succeeds.

Reply:
Updated now.